sync/src/types/mod.rs

```diff
@@ -442,6 +442,15 @@ impl From<&core::HeaderView> for BlockNumberAndHash {
     }
 }
 
+impl From<core::HeaderView> for BlockNumberAndHash {
+    fn from(header: core::HeaderView) -> Self {
+        Self {
+            number: header.number(),
+            hash: header.hash(),
+        }
+    }
+}
+
```

Comment:
We already have `impl From<&core::HeaderView> for BlockNumberAndHash` on above.

How about use `&` to convert `HeaderView` to `BlockNumberAndHash`?

like:
```rust
        let m_right: BlockNumberAndHash =
            (&self.get_ancestor(&m_right.hash(), m_left.number())?).into();

```

Reply:
Or 
```rust
        m_right = self.get_ancestor(&m_right.hash(), m_left.number()).as_ref()?.into();
```

Reply:
It's for usability, both the owned and reference versions of the `HeaderView` struct are commonly used and conversions between them are expected to be frequent in our codebase, implementing both `From` conversions can make the code more ergonomic and user-friendly. It allows our code to convert between the two versions using the same syntax and idioms, regardless of whether they have an owned or a reference version of the struct.